### DISCUSSION TITLE
How to send a command and then get the execution result synchronously

### DISCUSSION THREAD

**Discussion by chenqian-dev** [termux/termux-app#3642] [Q&A] [answered]
How to send a command and then get the execution result synchronously. For example, proot-distro login debian, how to get the login result from the Android layer? Because I need to wait for success and then perform some operations on the android layer.

**Comment by agnostic-apollo** [comment]
Send command from where and receive result where?

You probably need https://github.com/termux/termux-app/wiki/RUN_COMMAND-Intent if a separate android app.

**Reply by chenqian-dev** [reply]
If RUN_COMMAND-Intent is used, the process session will closed after the task is completed, which is not what I want;
My usage scenario is as follows: create a termux session, interact with it, and follow all commands, such as:
execute "proot-distro install debian", get the results synchronously, while the termux session needs to be kept and not closed because there are other steps later, such as
Execute "proot-distro login debian", etc.;
Therefore, my requirement is: open a termux session (it's a bash process?), maintain this session, each subsequent command will be executed synchronously in this session, and then get the result, and then judge the execution based on the result.
What should I do, thank you

**Reply by agnostic-apollo** [reply, marked as answer]
You can neither directly inject commands into running sessions with `RUN_COMMAND` or get back results before a session finishes. And for terminal sessions, termux returns the session transcript, and not the stdout or stderr streams separately, and it will contain user inputs as well. Parsing a transcript vs output streams will be different . What you can possibly do is send a `bash` script to termux that will contain all the commands that need to be executed, like to `bash -c`.

Another way might be to run a script that waits for a file to be created that contains commands to be executed in the main session, like with a `test -f` for loop with a sleep+timeout. You can write to the file to send commands from your app with background sessions. Once executed, you can send desired output/result back to your app with an intent with `am` command or if too large, then make your app run another background session that waits in a for loop with sleep and timeout to wait for a file to be created by the main session script with the result.

**Reply by chenqian-dev** [reply]
thanks a lot

**Reply by agnostic-apollo** [reply]
Welcome.
